.github/workflows/docs.yml

```
@@ -0,0 +1,94 @@
+name: Generate and Preview Rust Docs
+
+on:
+  pull_request:
+    branches:
+      - main  # Only generate docs for PRs targeting main
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  generate-docs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          path: zephyr-rust-lang
```

Why not use the module path?

```suggestion
          path: modules/lang/rust
```

---

I'm actually not sure it needs any path, since it seems to result in it coming out in zephyr-rust-lang/zephyr-rust-lang. But, now that the module is "real", I wonder if this can just update the module instead of checking it out somewhere else. I'll investigate.